### PR TITLE
Middleware added to redirect the user if the trial site expires

### PR DESCRIPTION
**Pull request by marslanabdulrauf**
** Description: **
This PR adds a middle to redirect user to pricing page if the trial site expires

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-4882

### CODE REVIEW

File: ecommerce/extensions/edly_ecommerce_app/middleware.py [hunk 1]
```diff
@@ -54,6 +54,19 @@ def process_request(self, request):
         if request.user.is_superuser or request.user.is_staff:
             return
 
+        current_site = get_current_site(request)
+        try:
+            current_site_configuration = current_site.siteconfiguration
+        except SiteConfiguration.DoesNotExist:
+            current_site_configuration = None
+            logger.warning('Site (%s) has no related site configuration.', current_site)
```
Comment (taimoor-ahmed-1) [inline ecommerce/extensions/edly_ecommerce_app/middleware.py:62]: kindly remove the log output as it bloats the logs for sample sites.

File: ecommerce/extensions/edly_ecommerce_app/middleware.py [hunk 2]
```diff
@@ -54,6 +54,19 @@ def process_request(self, request):
...
+
+        if current_site_configuration:
+            django_override_settings = current_site_configuration.get_edly_configuration_value('DJANGO_SETTINGS_OVERRIDE', {})
+            if django_override_settings.get('CURRENT_PLAN') == 'trial expired' and not _is_logged_in_path(request.path):
```
Comment (taimoor-ahmed-1) [inline ecommerce/extensions/edly_ecommerce_app/middleware.py:66]: we should add `trial expired` in constants and use the constant here.

**Review by taimoor-ahmed-1 — CHANGES_REQUESTED** [review]

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM

**Review by madnihamza1841 — APPROVED** [review]
LGTM

**Pull request merged**